### PR TITLE
Restructure project directory setup

### PR DESCRIPTION
**Pull request by drahnr**
Tackles #8

From an external point of view, there is only one minor difference:

`75bb239` introduces the trait `std::str::FromStr` for `Compressor` which was previously just a regular impl of `pub fn from_str(..)` - clippy complaint.

Hence this changeset mandates (at least) a minor release version.

### CODE REVIEW

File: src/rpm/headers/mod.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+mod header;
+mod lead;
+mod types;
+
+pub use header::*;
+pub use lead::*;
+pub use types::*;
+
+#[cfg(feature = "signing-meta")]
```
Comment (Richterrettich) [inline src/rpm/headers/mod.rs:9]: I am not sure about this cfg tag here. We use the SignatureHeaderBuilder in our RPMBuilder struct. Does this still work when this is gated off behind a cfg feature?

File: src/rpm/headers/mod.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+mod signature_builder;
```
Comment (Richterrettich) [inline src/rpm/headers/mod.rs:10]: I think this should be just the module `signature`.  This module may contain more functions for header signatures in the future

File: src/rpm/signature.rs [hunk 3]
```diff
@@ -1,7 +1,6 @@
+// re-export
```
Comment (Richterrettich) [inline src/rpm/signature.rs:1]: Do we really need this re export any more?

Comment (Richterrettich) [reply src/rpm/signature.rs:1]:
```rust
use rpm::signature::crypto::pgp::{Signer,Verifier};
```
does not seem much more ergonomic than
```rust
use rpm::crypto::pgp::{Signer,Verifier};
```

I think the clearest would be 

```rust
use rpm::signature::pgp::{Signer,Verifier};
```
All other paths should be deprecated (if public) and removed before 1.0.0

File: src/rpm/signature.rs [hunk 4]
```diff
@@ -1,7 +1,6 @@
...
 #[cfg(feature = "signing-pgp")]
```
Comment (Richterrettich) [inline src/rpm/signature.rs:2]: the feature gate should be `signing-pgp` for the whole export, but for the pgp export specifically

**Comment by Richterrettich** [comment]
Since it is not yet in 1.0.0 state, I am more liberal with breaking changes. If possible with a deprecation time of the old API. 
So the inclusion of `std::str::FromStr` is fine for me.
I like the overall structure you have proposed here.

The only thing we should discuss is the accessibility of the whole signature stuff, as stated in my comments.
The `crypto` crate should be private in my opinion. The word "crypto" is very generic and less focused than "signature" which is a much more specialized part of cryptography - and more fitting for the RPM use case.

**Comment by drahnr** [comment]
Intention Here was to keep the public export of modules identical, but I am happy to clean things up rename crypto to signature.

File: src/rpm/headers/mod.rs (hunk 1 above)
Comment (drahnr) [reply src/rpm/headers/mod.rs:9]: I am going to double check this.

**Comment by drahnr** [comment]
Renames feature flags from `signing-` to `signature-*` which is more descriptive - it provides both signing and verification of signatures.
Also moves crypto to signature into the `rpm` folder.

**Comment by Richterrettich** [comment]
It looks really good now. The command `cargo test --features test-with-podman` has some issues though:
```bash
error[E0433]: failed to resolve: could not find `test` in `signature`
  --> src/compat_tests.rs:34:50
   |
34 |         let (signing_key, _) = crate::signature::test::load_asc_keys();
   |                                                  ^^^^ could not find `test` in `signature`

error[E0433]: failed to resolve: could not find `test` in `signature`
   --> src/compat_tests.rs:107:65
    |
107 |         let (signing_key, verification_key) = crate::signature::test::load_asc_keys();
    |                                                                 ^^^^ could not find `test` in `signature`
```

can you have a look at this?

**Pull request merged**